### PR TITLE
API-4339: Get topic by topicName and ClientId endpoint added with tests

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/uk/gov/hmrc/pushpullnotificationsapi/repository/TopicsRepository.scala [hunk 1]
```diff
@@ -56,5 +58,10 @@ class TopicsRepository @Inject()(mongoComponent: ReactiveMongoComponent)
       case e: WriteResult if e.code.contains(MongoErrorCodes.DuplicateKey) =>
         Future.failed(DuplicateTopicException(s"${topic.topicName} ${topic.topicCreator.clientId}"))
     }
+
+  def getTopicByNameAndClientId(topicName: String, clientId: String)(implicit executionContext: ExecutionContext): Future[List[Topic]] = {
+    Logger.info(s"Getting topic by topicName:$topicName & clientId:$clientId")
+    find("topicName" -> topicName, "topicCreator.clientId" -> clientId)
```
Comment (lburgos) [inline app/uk/gov/hmrc/pushpullnotificationsapi/repository/TopicsRepository.scala:64]: should we do something like collection.find(query).one[Topic]? There's a unique index on these two parameters so there can only exist one item or zero. Using "find one" gives you an option with either a topic or none.

**Comment by lburgos** [comment]
I'm not sure about the query. Other than that looks good to me.

File: app/uk/gov/hmrc/pushpullnotificationsapi/repository/TopicsRepository.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsapi/repository/TopicsRepository.scala:64]: I'll look into it. I did try this 1st but the find method from collection that allows the one is deprecated. This is using the hmrc mogo find.

Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/pushpullnotificationsapi/repository/TopicsRepository.scala:64]: Have tried to do this but mixing Hmrc implementations with Reactive mongo and Play Json Readers / Writers is massively complicated. Going to leave this for now

**Pull request merged**